src/main/java/com/webank/wecross/stub/bcos/BCOSDriver.java

```diff
@@ -1048,6 +1062,110 @@ public void asyncGetBlock(
+            resource = proxyResult.getValue2();
+            input = Numeric.toHexString(proxyResult.getValue3());
+            methodId = input.substring(0, FunctionUtility.MethodIDWithHexPrefixLength);
+            if (logger.isDebugEnabled()) {
+                logger.debug("  resource: {}, methodId: {}", resource, methodId);
+            }
+        } else if (proxyInput.startsWith(
+                functionEncoder.buildMethodId(FunctionUtility.ProxySendTransactionTXMethod))) {
+            Tuple6<String, String, BigInteger, String, String, byte[]> proxyInputResult =
+                    FunctionUtility.getSendTransactionProxyFunctionInput(proxyInput);
+            xaTransactionID = proxyInputResult.getValue2();
+            path = proxyInputResult.getValue4();
+            resource = Path.decode(path).getResource();
+            String methodSig = proxyInputResult.getValue5();
+            methodId = functionEncoder.buildMethodId(methodSig);
+
+            if (logger.isDebugEnabled()) {
+                logger.debug(
+                        "xaTransactionID: {}, path: {}, methodSig: {}, methodId: {}",
+                        xaTransactionID,
+                        path,
+                        methodSig,
+                        methodId);
+            }
+        } else {
+            // transaction not send by proxy
+            transaction.setTransactionByProxy(false);
+            block.getTransactionsWithDetail().add(transaction);
+            return;
+        }
+
+        transaction
+                .getTransactionRequest()
+                .getOptions()
+                .put(StubConstant.XA_TRANSACTION_ID, xaTransactionID);
+        transaction.setResource(resource);
+        // query ABI
+        String finalMethodId = methodId;
+        asyncCnsService.queryABI(
+                resource,
+                this,
+                connection,
+                (queryABIException, abi) -> {
+                    if (Objects.nonNull(queryABIException)) {
+                        logger.error(
+                                "Query abi failed, transactionHash: {}, e: ",
+                                transactionHash,
+                                queryABIException);
+                        block.getTransactionsWithDetail().add(transaction);
+                        return;
+                    }
+
+                    ABIDefinition function =
+                            abiDefinitionFactory
+                                    .loadABI(abi)
+                                    .getMethodIDToFunctions()
+                                    .get(finalMethodId);
+
+                    if (Objects.isNull(function)) {
+                        logger.warn(
+                                "Maybe abi is upgraded, Load function failed, methodId: {}",
+                                finalMethodId);
+
+                        block.getTransactionsWithDetail().add(transaction);
```

Comment:
Here is an asynchronous call, you should not set the value in the callback, it is possible that the function will return before the callback is executed.